lang/cs/Org.Apache.REEF.Client/YARN/RESTClient/YarnClientNoCancellationToken.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Org.Apache.REEF.Client.YARN.RESTClient.v1;
+using Org.Apache.REEF.Utilities.AsyncUtils;
+
+namespace Org.Apache.REEF.Client.YARN.RESTClient
+{
+    internal sealed partial class YarnClient
+    {
+        public async Task<ClusterInfo> GetClusterInfoAsync()
```

Comment:
The methods are internal as the class visibility takes precedence. The methods are public because if the class is made public all the methods are intended to be public. It is a common pattern.
Take a look at:
http://stackoverflow.com/questions/711361/public-vs-internal-methods-on-an-internal-class
Specifically: http://stackoverflow.com/a/711384/285199